optaplanner-wb-screens/optaplanner-wb-guided-rule-editor/optaplanner-wb-guided-rule-editor-client/src/main/java/org/optaplanner/workbench/screens/guidedrule/client/plugin/BendableHardConstraintMatchRuleModellerActionPlugin.java

```diff
@@ -0,0 +1,114 @@
+public class BendableHardConstraintMatchRuleModellerActionPlugin implements RuleModellerActionPlugin {
+
+    private static final Set<String> SUPPORTED_SCORE_HOLDER_TYPES;
+
+    static {
+        SUPPORTED_SCORE_HOLDER_TYPES = new HashSet<>( 3 );
+        SUPPORTED_SCORE_HOLDER_TYPES.add( "org.optaplanner.core.api.score.buildin.bendable.BendableScoreHolder" );
+        SUPPORTED_SCORE_HOLDER_TYPES.add( "org.optaplanner.core.api.score.buildin.bendablelong.BendableLongScoreHolder" );
+        SUPPORTED_SCORE_HOLDER_TYPES.add( "org.optaplanner.core.api.score.buildin.bendablebigdecimal.BendableBigDecimalScoreHolder" );
+    }
+
+    @Inject
+    private Caller<ScoreHolderService> scoreHolderService;
+
+    @Inject
+    private ActionPluginUtils actionPluginUtils;
+
+    public BendableHardConstraintMatchRuleModellerActionPlugin() {
+    }
+
+    @Override
+    public boolean accept( final IAction iAction ) {
+        return iAction instanceof ActionBendableHardConstraintMatch;
+    }
+
+    @Override
+    public void addPluginToActionList( final Path projectPath,
+                                       final Command addCommand ) {
+        actionPluginUtils.addPluginToActionList( projectPath,
+                                                 addCommand,
+                                                 SUPPORTED_SCORE_HOLDER_TYPES );
+    }
+
+    @Override
+    public IAction createIAction( final Path projectPath ) {
+        return new ActionBendableHardConstraintMatch();
+    }
+
+    @Override
+    public String getId() {
+        return "BENDABLE_HARD_CONSTRAINT_MATCH";
+    }
+
+    @Override
+    public String getActionAddDescription() {
+        // TODO i18n
+        return "Modify Hard Score";
+    }
+
+    @Override
+    public RuleModellerWidget createWidget( final RuleModeller ruleModeller,
+                                            final EventBus eventBus,
+                                            final IAction iAction,
+                                            final Boolean readOnly ) {
+        BendableConstraintMatchRuleModellerWidget widget = new BendableConstraintMatchRuleModellerWidget( ruleModeller,
+                                                                                                          eventBus,
+                                                                                                          (ActionBendableHardConstraintMatch) iAction,
+                                                                                                          readOnly,
+                                                                                                          "Hard Score" );
+        actionPluginUtils.initScoreHolderAwarePlugin( ruleModeller.getPath(),
+                                                      widget,
+                                                      SUPPORTED_SCORE_HOLDER_TYPES );
+
+        scoreHolderService.call( new RemoteCallback<Collection<BendableScoreLevelsWrapper>>() {
```

Comment:
So to initialise a single plugin it looks like we do **three** server round-trips; and given you have a number of plugins; opening a Guided Rule Editor instance could be come pretty slow (on a _real_ environment; i.e. not ```localhost```)